aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/actions/AuthenticationCognitoActions.kt

```diff
@@ -33,10 +34,25 @@ object AuthenticationCognitoActions : AuthenticationActions {
         Action<AuthEnvironment>("ConfigureAuthN") { id, dispatcher ->
             logger?.verbose("$id Starting execution")
             val userPoolTokens = event.storedCredentials?.cognitoUserPoolTokens
-            val evt = userPoolTokens?.let {
-                val signedInData = SignedInData("", "", Date(), SignInMethod.SRP, it)
-                AuthenticationEvent(AuthenticationEvent.EventType.InitializedSignedIn(signedInData))
-            } ?: AuthenticationEvent(AuthenticationEvent.EventType.InitializedSignedOut(SignedOutData()))
+            val evt = if (userPoolTokens?.accessToken != null) {
+                AuthenticationEvent(
+                    AuthenticationEvent.EventType.InitializedSignedIn(
+                        SignedInData(
+                            JWTParser.getClaim(
```

Comment:
is `SignedInData("","", ..)` a valid case?

Reply:
This is how it is currently implemented so wanted to keep it as is.

Reply:
Guest access will have no access tokens and hence the userid and username will be null

Reply:
Understandable, this code block is run only for `accessToken` is not null.
Is it implied that when userId and username are "", it is a "Guest User" scenario?

Reply:
Now that I think about it, we do not actually care what is submitted here as we parse it when a getCurrentUser is called. I am going to revert it to always send "", "" as that value is not used at all.